server/stream.go

```diff
@@ -5456,7 +5456,9 @@ func (mset *stream) processJetStreamMsg(subject, reply string, hdr, msg []byte,
 		}
 
 		// Check to see if we are over the max msg size.
-		if int32(len(hdr)+len(msg)) > mset.srv.getOpts().MaxPayload {
+		// Subtract to prevent against overflows.
+		maxPayload := mset.srv.getOpts().MaxPayload
+		if int32(len(hdr)) > maxPayload || int32(len(msg)) > maxPayload-int32(len(hdr)) {
```

Comment:
I'm wondering whether using `int64` here would be safer, i.e.
```
maxPayload := int64(mset.srv.getOpts().MaxPayload)
hdrLen := int64(len(hdr))
msgLen := int64(len(msg))

if hdrLen > maxPayload || msgLen > maxPayload-hdrLen {
    // overflow or limit exceeded
}
```

Reply:
Done